### ISSUE TITLE
Japanese characters encoding breaks resource search

### ISSUE THREAD

**Issue by WillyReyno** [laravel/nova-issues#5184]
- Laravel Version: 9.43.0
- Nova Version: 4.19.5
- PHP Version: 8.0.20
- Database Driver & Version: MySQL 8.0
- Operating System and Version: MacOS Ventura 13.0.1
- Browser type and version: Chrome 108.0.5359.124

### Description:

Since migrating to Nova 4 I noticed a change in how Japanese characters within arrays are stored in database, which prevents searching them in resource index and global search.

#### Behavior before Nova 4

Before upgrading to Nova 4, Japanese characters in array were stored clearly like this: 
![Capture d’écran 2022-12-26 à 16 41 59](https://user-images.githubusercontent.com/5576482/209564681-0e9d17c6-d1ae-4df1-89cd-aca57a5941b2.png)

Searching `オレ` before Nova 4 was correctly returning the resource: 
![Capture d’écran 2022-12-26 à 16 36 04](https://user-images.githubusercontent.com/5576482/209564409-57697bc4-d6f0-4e65-bbe1-9178e52378e9.png)
_"Names" column only shows the first item of the array_

#### Behavior since Nova 4

Now, Japanese texts within arrays are stored encoded. So If I edit my resource and add a new `"test"` item to the array, it would result in the following in database: 

![Capture d’écran 2022-12-26 à 16 40 09](https://user-images.githubusercontent.com/5576482/209564507-072576e2-c0ea-4f9b-8c96-890f7b7d6415.png)

And search won't work anymore for the Japanese item: 

![Capture d’écran 2022-12-26 à 16 37 45](https://user-images.githubusercontent.com/5576482/209564523-12fea2af-38ac-4ea0-a6f1-f415fa91ca04.png)

While still working for the "test" one: 
![Capture d’écran 2022-12-26 à 16 41 15](https://user-images.githubusercontent.com/5576482/209564608-756b9e82-fc62-45af-91f6-5b32ab3e77cb.png)

### Detailed steps to reproduce the issue on a fresh Nova installation:

* Create a `text` column on any table
* In the model, cast it as array : `protected $casts = ['your_column']`
* Add a `Code` field to display your array on the Nova resource,
* Create a resource and add the following array to your column's code field : `["オレ", "test"]`
* On the resource, declare `your_column` as a SearchableJson : 
```
    public static function searchableColumns()
    {
        return [
            'id',
            new SearchableJson('your_column')
        ];
    }
```
* Try searching for `test` on the resource index, your resource should be returned
* Try searching for `オレ` on the resource index, the resource will not be returned

Feel free to ask me if something is not clear!

Thanks for your help

**Comment by crynobone** [comment]
`SearchableJson` is a new feature for Laravel Nova 4 and has no equivalent in Nova 3.

**Comment by WillyReyno** [comment]
Oh indeed, I didn't think about that. But without even mentioning Nova 3, the fact that the old way of storing Japanese characters in database works with `SearchableJson` and not the new encoded is still an issue, right?

I will make a reproducing repository later today!

**Comment by crynobone** [comment]
> the fact that the old way of storing Japanese characters in database works with SearchableJson and not the new encoded is still an issue, right?

There no code changes to `Code` field and we don't have any known changes to "encoded" claims. Will need a reproducing repository.

**Comment by WillyReyno** [comment]
I found my "issue" while creating the reproducing repository: my database columns were previously json typed, I changed that to text someday which made the "encoding" change for all Japanese texts in that column. I don't remember why I did that, but I did it on the same day I upgraded to Nova 4, which lead me to think it was an upgrade issue.

Rolled back to json typed columns, search works fine.

The fact that Japanese characters are not searchable within text columns is still there, but I don't know if it's really something that can be addressed. 

Thank you and sorry to have wasted your time
